axelor-sale/src/main/java/com/axelor/apps/sale/service/cartline/CartLineRetrievalServiceImpl.java

```diff
@@ -37,8 +37,10 @@ public CartLineRetrievalServiceImpl(CartLineRepository cartLineRepository) {
   public CartLine getCartLine(Cart cart, Product product) {
     return cartLineRepository
         .all()
-        .filter("self.product = :product AND self.cart = :cart")
+        .filter(
+            "(self.product = :product OR self.variantProduct=:variantProduct) AND self.cart = :cart")
         .bind("product", product)
+        .bind("variantProduct", product)
```

Comment:
Does this filter work in a Cart with several CartLine with the same product but different variant ?

Reply:
Yes, it works